packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -1,21 +1,246 @@
 import React, { useState, useEffect, useCallback, useRef } from "react"
```

Comment:
This file is a bit too big for my taste, any chance we can split it up?

Reply:
I agree. I pulled out all of the components to separate files and cleaned everything up.